isis/src/base/objs/Chip/unitTest.cpp

```diff
@@ -134,15 +134,15 @@ int main() {
   cout << "Test load chip from cube with rotation and clipping polygon " << endl;
   chip.TackCube(26.0, 25.0);
 
-  geos::geom::CoordinateSequence *pts = new geos::geom::CoordinateArraySequence();
+  geos::geom::CoordinateArraySequence *pts = new geos::geom::CoordinateArraySequence();
   pts->add(geos::geom::Coordinate(23.0, 22.0));
   pts->add(geos::geom::Coordinate(28.0, 22.0));
   pts->add(geos::geom::Coordinate(28.0, 27.0));
   pts->add(geos::geom::Coordinate(25.0, 28.0));
   pts->add(geos::geom::Coordinate(23.0, 22.0));
-  vector<geos::geom::Geometry *> polys;
+  vector<geos::geom::Geometry *> *polys = new vector<geos::geom::Geometry *>;
```

Comment:
I think this results in a memory leak.  We could use a smart pointer via std::make_unique or std::make_shared or just track it down and delete the memory

Reply:
I added `delete polys;` but this caused both tests to fail.

Reply:
That's interesting... is there something that I'm missing with the heap allocation here?  I'd expect it to need deallocation somewhere unless `polys` is part of an object with its own destructors.

Maybe smart pointers would manage this for us?  A memory leak in the tests isn't great, but it's also not the end of the world. @Kelvinrr @acpaquette any thoughts?

Reply:
ISIS handles pointers really poorly and likely needs to have verification that pointers get cleaned up. In this case the chip classes destruction deletes the pointer see here:
https://github.com/DOI-USGS/ISIS3/blob/1e21b588f16096b1a7ad4e7ddd597ba1500a876e/isis/src/base/objs/Chip/Chip.cpp#L89